### PR TITLE
Update examples and tutorial after automatic sharding has landed

### PR DESCRIPTION
**Pull request by NivekT**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #505
* #504

After the `DataLoader` automatic sharding for DataPipe PR has landed in core, this PR is updating the relevant example and tutorial

This should be cherry-picked to release.

Differential Revision: [D37040383](https://our.internmc.facebook.com/intern/diff/D37040383)

### CODE REVIEW

**Review by VitalyFedyunin — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -200,10 +200,12 @@ convert it to `MapDataPipe` as needed.
 
 Q: How is multiprocessing handled with DataPipes?
 
-A: Multi-process data loading is still handled by DataLoader, see the
+A: Multi-process data loading is still handled by the `DataLoader`, see the
 [DataLoader documentation for more details](https://pytorch.org/docs/stable/data.html#single-and-multi-process-data-loading).
-If you would like to shard data across processes, use `ShardingFilter` and provide a `worker_init_fn` as shown in the
-[tutorial](https://pytorch.org/data/beta/tutorial.html#working-with-dataloader).
+As of PyTorch version 1.12.0 (TorchData version 0.4.0), data sharding is automatically done for DataPipes within the
```
Comment (ejguan) [inline README.md:205]: nit: As current nightly has bumped to 1.13.0 and 0.5.0, Would it be better to state the version of PyTorch >= 1.12.0 and TorchData >= 0.4.0?

**Review by ejguan — APPROVED** [review]
Thank you so much for keeping our document updated

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37040383).

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37040383).

**Pull request closed without merging**